themes/default/content/blog/kubernetes-architecture-templates/index.md

```diff
@@ -0,0 +1,388 @@
+    """,
+        },
+        opts=pulumi.ResourceOptions(provider=new_provider)
+    )
+
+    # Create a Deployment with a user-defined number of replicas
+    webserverdeployment = kubernetes.apps.v1.Deployment(
+        "webserverdeployment",
+        metadata=kubernetes.meta.v1.ObjectMetaArgs(
+            namespace=webserverns.metadata.name,
+        ),
+        spec=kubernetes.apps.v1.DeploymentSpecArgs(
+            selector=kubernetes.meta.v1.LabelSelectorArgs(
+                match_labels=app_labels,
+            ),
+            replicas=num_replicas,
+            template=kubernetes.core.v1.PodTemplateSpecArgs(
+                metadata=kubernetes.meta.v1.ObjectMetaArgs(
+                    labels=app_labels,
+                ),
+                spec=kubernetes.core.v1.PodSpecArgs(
+                    containers=[kubernetes.core.v1.ContainerArgs(
+                        image="nginx",
+                        name="nginx",
+                        volume_mounts=[kubernetes.core.v1.VolumeMountArgs(
+                            mount_path="/etc/nginx/nginx.conf",
+                            name="nginx-conf-volume",
+                            read_only=True,
+                            sub_path="nginx.conf",
+                        )],
+                    )],
+                    volumes=[kubernetes.core.v1.VolumeArgs(
+                        config_map=kubernetes.core.v1.ConfigMapVolumeSourceArgs(
+                            items=[kubernetes.core.v1.KeyToPathArgs(
+                                key="nginx.conf",
+                                path="nginx.conf",
+                            )],
+                            name=webserverconfig.metadata.name,
+                        ),
+                        name="nginx-conf-volume",
+                    )],
+                ),
+            ),
+        ),
+        opts=pulumi.ResourceOptions(provider=new_provider)
+    )
+
+    # Expose the Deployment as a Kubernetes Service
+    webserverservice = kubernetes.core.v1.Service(
+        "webserverservice",
+        metadata=kubernetes.meta.v1.ObjectMetaArgs(
+            namespace=webserverns.metadata.name,
+        ),
+        spec=kubernetes.core.v1.ServiceSpecArgs(
+            ports=[kubernetes.core.v1.ServicePortArgs(
+                port=80,
+                target_port=80,
+                protocol="TCP",
+            )],
+            selector=app_labels,
+            type="LoadBalancer"
+        ),
+        opts=pulumi.ResourceOptions(provider=new_provider)
+    )
+
+    # Export some values for use elsewhere
+    pulumi.export("deploymentName", webserverdeployment.metadata.name)
+    pulumi.export("serviceName", webserverservice.metadata.name)
+    return webserverservice.status.apply(lambda val: val.load_balancer.ingress[0].ip)
+```
+
+Now that all our changes have been made, let's try running it! We'll need to append `poetry run` to the commands, unless you choose to switch to the Poetry shell ahead of time:
```

Comment:
It should be "prepend" instead of "append", I believe.

Reply:
Good catch!